content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+It follows the idea of having some neglectable error when reaching of consensus as Nakamoto suggested, which makes it probabilistic and means there is no 100% certainty of reaching a consensus.
+
+The number of messages each node has to handle per decision is with O(k) complexity and does not grow as the network scales up, consequently the system can scale to millions of validators all participating in the consensus to achieve a global size decentralized blockchain network.
+
+Logically, Avalanche is a validator protocol. Each node has a known number of validators (other nodes) called _validator list_.
+Each of the validators stakes some amount in the network and the probability of their selection in the lists of other nodes is weighted by this staked amount.
+Validators listen for transactions.
+When they see a new transaction they vote on whether it is to be `Accepted` or `Rejected`.
+Transactions that appear virtuous are voted on as `Accepted` by a validator, and validators are always expected to have deterministic behavior.
+If the transaction is conflicting with the previous one that the validator had voted for, is is rejected.
+Each validator is a completely independent decision maker and there is no leader election.
+
+However, by protocol, each node uses the same process to determine whether a decision is virtuous and how likely they are in consensus with the rest of the network.
+This process is referred to as `repeated random subsampling` and it determines whether a transaction is preferred and that the rest of the network agrees on this it.
+At a high-level, a validator randomly selects other validators to ask them about their current statement.
+Procedure repeats multiple times on new randomly selected nodes until the node builds up enough data to determine that the probability of being correct is high enough that it can be considered impossible to consider the opposite value.
+
+In order to perform the consensus algorithm like this, Avalanche contains of multiple single decree Snowball instances instantiated as a multi-decree protocol that maintains a dynamic, append-only Directed Acyclic Graph (DAG) of all known transactions.
+Maintaining a DAG improves efficiency, because a single vote on a DAG vertex implicitly votes for all transactions on the path to the genesis vertex.
+It also improves security, because the DAG intertwines the fate of transactions, similar to the Bitcoin blockchain, which renders past decisions difficult to undo without the approval of correct nodes.
+
+When a client creates a transaction, it names one or more parent transactions, which are included inseparably in the inputs and their connections form the edges of the DAG.
+This parent-child relationship encoded in the DAG may or may not correspond to application specific dependencies — for example a child transaction need not spend to have relationship with the funds received in parent transaction.
```

Comment:
`need not spend funds received in a parent transaction`

Reply:
This sentence has been removed when cutting out details in the information.